mbed_greentea/mbed_greentea_cli.py

```diff
@@ -726,140 +645,116 @@ def main_cli(opts, args, gt_instance_uuid=None):
         shuffle_random_seed = round(float(opts.shuffle_test_seed), SHUFFLE_SEED_ROUND)
 
     ### Testing procedures, for each target, for each target's compatible platform
-    for yotta_target_name in yt_target_platform_map:
-        gt_logger.gt_log("processing '%s' yotta target compatible platforms..."% gt_logger.gt_bright(yotta_target_name))
-
-        for platform_name in yt_target_platform_map[yotta_target_name]:
-            gt_logger.gt_log("processing '%s' platform..."% gt_logger.gt_bright(platform_name))
-
-            ### Select MUTS to test from list of available MUTS to start testing
-            mut = None
-            number_of_parallel_instances = 1
-            for mbed_dev in ready_mbed_devices:
-                if accepted_target_ids and mbed_dev['target_id'] not in accepted_target_ids:
-                    continue
-
-                if mbed_dev['platform_name'] == platform_name:
-                    mut = mbed_dev
-                    muts_to_test.append(mbed_dev)
-                    gt_logger.gt_log("using platform '%s' for test:"% gt_logger.gt_bright(platform_name))
-                    for k in mbed_dev:
-                        gt_logger.gt_log_tab("%s = '%s'"% (k, mbed_dev[k]))
-                    if number_of_parallel_instances < parallel_test_exec:
-                        number_of_parallel_instances += 1
-                    else:
-                        break
-
-            # Configuration print mode:
-            if opts.verbose_test_configuration_only:
+    for test_build in test_spec.get_test_builds():
+        gt_logger.gt_log("processing target '%s' toolchain '%s' compatible platforms..." %
+                         (gt_logger.gt_bright(test_build.get_platform()),
+                                             gt_logger.gt_bright(test_build.get_toolchain())))
+
+        platform_name = test_build.get_platform()
```

Comment:
Minor point you are invoking test_build.get_platform() twice here , would be more efficient to move this line above the gt_logger line  and then use platform_name as the parameter?